projects/ng-event-plugins/src/events/longtap.event.ts

```diff
@@ -0,0 +1,142 @@
+import type {EventManager} from '@angular/platform-browser';
+
+import type {EventManagerPlugin} from '../types/event-manager-plugin';
+import {LongtapEvent} from '../types/longtap.event';
+import {isIos} from '../utils/is-ios';
+
+const TAP_DELAY = 700;
+const SAFE_NAVIGATOR = typeof navigator === 'undefined' ? null : navigator;
+const MOVE_THRESHOLD = 15;
+
+export class LongtapEventPlugin implements EventManagerPlugin {
+    private readonly isIOS = !!SAFE_NAVIGATOR && isIos(SAFE_NAVIGATOR);
+
+    public readonly manager!: EventManager;
+
+    public addEventListener(
+        element: HTMLElement,
+        _event: string,
+        handler: (event: Event) => void,
+    ): Function {
+        const removeLongtapEventPolyfill = this.isIOS
+            ? this.listenTouchEvents(element)
+            : this.listenContextmenuEvent(element);
+
+        element.addEventListener('longtap', handler);
+
+        return () => {
+            removeLongtapEventPolyfill();
+
+            element.removeEventListener('longtap', handler);
+        };
+    }
+
+    public supports(event: string): boolean {
+        return /^longtap$/i.test(event);
+    }
+
+    private listenContextmenuEvent(element: HTMLElement): Function {
+        return this.manager.addEventListener(
+            element,
+            'contextmenu.prevent.stop',
+            ({clientX, clientY}: MouseEvent) => {
+                this.dispatchLongtapEvent(element, clientX, clientY);
+            },
+        );
+    }
+
+    private listenTouchEvents(element: HTMLElement): Function {
+        let longTapTimeout: any = null;
+        let touchStartCoords: {
+            clientX: number;
+            clientY: number;
+        } | null = null;
+
+        function reset(): void {
```

Comment:
Should it be an arrow function for consistency? I don't think we ever used nested functions like that.